src/fields/multi_field.impl.hh

```diff
@@ -130,25 +123,29 @@ bool MultiField<spacedim, Value>::is_constant(Region reg) {
 
 
 template<int spacedim, class Value>
-void MultiField<spacedim, Value>::set_up_components() {
-	ASSERT(this->shared_->comp_names_.size(), "Vector of component names is empty!\n");
+void MultiField<spacedim, Value>::setup_components() {
+	unsigned int comp_size = this->shared_->comp_names_.size();
+	string full_name;
+	ASSERT(comp_size, "Vector of component names is empty!\n");
 	ASSERT(this->shared_->mesh_, "Mesh is not set!\n");
 
-    sub_fields_.resize( this->shared_->comp_names_.size() );
-    for(unsigned int i_comp=0; i_comp < size(); i_comp++)
+    sub_fields_.reserve( comp_size );
+    for(unsigned int i_comp=0; i_comp < comp_size; i_comp++)
     {
+    	if (this->shared_->comp_names_[i_comp].length() == 0)
+    		full_name = name();
+    	else {
+    		full_name = this->shared_->comp_names_[i_comp] + "_" + name();
+    	}
+
+    	sub_fields_.push_back( SubFieldType(i_comp, name(), full_name) );
```

Comment:
I see, here the copy constructor for Field<>should copy name_as well. So we have to check all places where this constructor is used.